docs/Contributors_Guide/directory_tree.rst

```diff
@@ -0,0 +1,74 @@
+**************
+Directory Tree
+**************
+
+Goals
+=====
+* No conflicts between different packages in their external packages
+* No circular dependencies
+* If a user/dev introduces a new dependency they need to add an entry in environment.yaml and setup.cfg 
+
+Action Items
+============
+Need GitHub Actions that test the setup.cfg and environment.yml to ensure that our code runs within
+Python and follows this type of generic structure:
+
+Python structure::
+
+   pkg
+   ├── __init__.py
+   ├── module1.py
+   └── subpkg
+     ├── __init__.py
+     └── module2.py
```

Comment:
what is `pkg` and `subpkg`?  Are we going to allow installing just a `subpkg` from within a `pkg`?